features/step_definitions/trigger.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const sdapi = require('../support/sdapi');
+const request = require('../support/request');
+const { Before, Given, When, Then } = require('cucumber');
+
+const TIMEOUT = 240 * 1000;
+
+Before({
+    tags: '@trigger'
+}, function hook() {
+    this.repoOrg = this.testOrg;
+    this.repoName = 'functional-trigger';
+    this.pipelineIdA = null;
+    this.pipelineIdB = null;
+    this.successAJobId = null;
+    this.failAJobId = null;
+    this.successBJobId = null;
+    this.failBJobId = null;
+    this.parallelAJobId = null;
+    this.parallelB1JobId = null;
+    this.parallelB2JobId = null;
+    this.builds = null;
+});
+
+Given(/^an existing pipeline on branch "(.*)" with the workflow jobs:$/, {
+    timeout: TIMEOUT
+}, function step(branch, table) {
+    let pipelineVarName = 'pipelineIdA';
+
+    if (branch === 'pipelineB') {
+        pipelineVarName = 'pipelineIdB';
+    }
+
+    return this.ensurePipelineExists({
+        repoName: this.repoName,
+        branch,
+        pipelineVarName,
+        table,
+        shouldNotDeletePipeline: true });
+});
+
+Given(/^an existing pipeline on branch "(.*)" with job "(.*)"$/, {
+    timeout: TIMEOUT
+}, function step(branch, jobName) {
+    let pipelineVarName = 'pipelineIdA';
+
+    if (branch === 'pipelineB') {
+        pipelineVarName = 'pipelineIdB';
+    }
+
+    return this.ensurePipelineExists({
+        repoName: this.repoName,
+        branch,
+        pipelineVarName,
+        jobName,
+        shouldNotDeletePipeline: true });
+});
+
+When(/^the "(.*)" job on branch "(?:.*)" is started$/,
+    { timeout: TIMEOUT }, function step(jobName) {
+        let jobId;
+        let buildVarName;
+
+        switch (jobName) {
+        case 'success_A':
```

Comment:
can we use hashmaps here instead so we don't need to have long switch like this

Reply:
one hashmap to map`fail_A` to `successAJobId` and `successABuildId`?

Reply:
updated